libs/coin-modules/coin-solana/src/preload.ts

```diff
@@ -17,18 +17,39 @@ import { getCALHash, setCALHash } from "./logic";
 import { addTokens, convertSplTokens } from "@ledgerhq/cryptoassets/tokens";
 import { AxiosError } from "axios";
 import { getCryptoAssetsStore } from "./cryptoAssetsStore";
-
-let shouldSkipTokenLoading = false;
-
-export function setShouldSkipTokenLoading(skip: boolean): void {
-  shouldSkipTokenLoading = skip;
-}
+import { resolveTokenAdder } from "@ledgerhq/coin-framework/crypto-assets/utils";
+import coinConfig from "./config";
+
+const storeTokens = (tokens: SPLToken[]) => {
+  const addTokensToStore = resolveTokenAdder(getCryptoAssetsStore);
+  if (addTokensToStore) {
+    const convertedTokens = tokens.map(convertSplTokens);
+    addTokensToStore(convertedTokens);
+  } else {
+    addTokens(tokens.map(convertSplTokens));
+  }
+};
```

Comment:
No big deal, just simpler
```suggestion
const storeTokens = (tokens: SPLToken[]) => {
  const addTokensToStore = resolveTokenAdder(getCryptoAssetsStore);
  const convertedTokens = tokens.map(convertSplTokens);
  if (addTokensToStore) {
    addTokensToStore(convertedTokens);
  } else {
    addTokens(convertedTokens);
  }
};
```